processing/src/main/java/io/druid/query/aggregation/AggregateCombiner.java

```diff
@@ -0,0 +1,66 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation;
+
+import io.druid.segment.ColumnValueSelector;
+
+/**
+ * AggregateCombiner is used to combine rollup aggregation results from serveral "rows" of different indexes during index
+ * merging (see {@link io.druid.segment.IndexMerger}).
+ *
+ * The state of the implementations of this interface is an aggregation value (either a primitive or an object), that
+ * could be queried via {@link ColumnValueSelector}'s methods. Before {@link #reset} is ever called on an
+ * AggregateCombiner, it's state is undefined and {@link ColumnValueSelector}'s methods could return something random,
+ * or null, or throw an exception.
+ *
+ * @see AggregatorFactory#makeAggregateCombiner()
+ * @see LongAggregateCombiner
+ * @see DoubleAggregateCombiner
+ * @see ObjectAggregateCombiner
+ */
+public interface AggregateCombiner extends ColumnValueSelector
+{
+  /**
+   * Resets this AggregateCombiner's state value to the value of the given selector, e. g. after calling this method
+   * combiner.get*() should return the same value as selector.get*().
+   *
+   * If the selector is an {@link io.druid.segment.ObjectColumnSelector}, the object returned from {@link
+   * io.druid.segment.ObjectColumnSelector#get()} must not be modified, and must not become a subject for modification
+   * during subsequent {@link #combine} calls.
+   */
+  void reset(ColumnValueSelector selector);
+
+  /**
+   * Combines this AggregateCombiner's state value with the value of the given selector and saves it in this
+   * AggregateCombiner's state, e. g. after calling combiner.combine(selector), combiner.get*() should return the value
+   * that would be the result of {@link AggregatorFactory#combine
+   * aggregatorFactory.combine(combiner.get*(), selector.get*())} call.
+   *
+   * Unlike {@link AggregatorFactory#combine}, if the selector is an {@link io.druid.segment.ObjectColumnSelector}, the
+   * object returned from {@link io.druid.segment.ObjectColumnSelector#get()} must not be modified, and must not become
+   * a subject for modification during subsequent combine() calls.
+   *
+   * Since the state of AggregateCombiner is undefined before {@link #reset} is ever called on it, the effects of calling
+   * combine() on an AggregateCombiner instance are also undefined in this case.
+   *
+   * @see AggregatorFactory#combine
+   */
+  void combine(ColumnValueSelector selector);
```

Comment:
Elsewhere this functionality is called `fold`, which to me indicates you are eliminating the prior state of this object, which is the behavior here. `combine` seems to indicate that it may or may not return a new object instead of folding in the other values. Would `fold` be more appropriate here?